crates/bencode/src/serde_bencode_de.rs

```diff
@@ -1,79 +1,96 @@
-use buffers::ByteBuf;
-use serde::de::Error as DeError;
-use sha1w::{ISha1, Sha1};
+use super::*;
 
 pub struct BencodeDeserializer<'de> {
-    buf: &'de [u8],
+    raw_buf: &'de [u8],
+    offset: usize,
     field_context: Vec<ByteBuf<'de>>,
     parsing_key: bool,
-
-    // This is a f**ing hack
-    pub is_torrent_info: bool,
-    pub torrent_info_digest: Option<[u8; 20]>,
 }
 
 impl<'de> BencodeDeserializer<'de> {
+    fn buf(&'de self) -> &'de [u8] {
```

Comment:
why change this? I didn't need to do it in my reference implementation.

The original code used one field - buf, now you have 2 of them + a method that you need not forget to call. But most importantly, it worked just fine, so why change it?

I suggest not changing things unrelated to the PR just because you don't like the code for some reason. It increases the surface area to review, makes the change questionable. In general if it ain't broken, I wouldn't change it.